File: C++Projects/POO_3964_Clase02_ClasesEnArchivos/Mamifero.cpp
//
// Created by Diego Cruz on 12/03/2024.
//

#include "Mamifero.h"

// Constructor de la clase
Mamifero::Mamifero(string nombre){
    this->nombre = nombre;
}

// Setter y Getter
string Mamifero::getNombre(){
    return nombre;
}
void Mamifero::setNombre(string nombre){
    this->nombre = nombre;
}

File: JavaProjects/POO_3964_Clase04_HerenciaMultiple/src/ParentClasses/Mamifero.java
package ParentClasses;

public class Mamifero {
    // Propiedades de la clase
    private String nombre;
    // Constructor
    public Mamifero(String nombre){
        this.nombre = nombre;
        System.out.println("\nHa creado un mamífero llamado " + nombre);
    }
    // Método propio de la clase "Mamifero"
    public void amamantar() {
        System.out.println("Tomo leche proveniente de mi madre.");
    }

    // Getter y Setter
    public String getNombre() {
        return nombre;
    }
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
}
